### DISCUSSION TITLE
How to use a setup file with global vars ?

### DISCUSSION THREAD

**Discussion by qlaffont** [tapjs/tapjs#1002] [Q&A] [answered]
I want to use a setup file as we can set it in jest (setupFiles).

Reason :
- [Start] Initiate my server before all tests
- [End] Close server, DB connections, clean DBs

**Comment by isaacs** [comment]
Would this be run in each test s
process, or a separate process that runs before the tests?

**Reply by qlaffont** [reply]
It needs to run one time before all tests and after all tests

**Reply by isaacs** [reply]
Oh then just do `--before=./path/to/script.js` or `tap config set before=./path/to/script.js` to persist it.

**Reply by isaacs** [reply]
But note, unlike jest, tap runs each test in its own isolated node process. So the before/after scripts will run to completion and exit. They can't eg set globals or have a server running persistently unless you spawn a background process. The `t-up` module can help with this, but I haven't used it in a while.

**Reply by qlaffont** [reply]
Thanks i will check t-up !

**Reply by isaacs** [reply]
Haha, actually that module is super dated and depends on a 3-majors-back version of tap, so maybe skip it.

Here's a super trimmed down example that should help, though: https://github.com/tapjs/tap-setup-script-example/tree/main

Depending on what you're doing, it might be more/less complicated in your situation, and maybe some of the types of guardrails t-up provides might be worthwhile, but I think it's not the end of the world to just hack something together for your tests. Often that's better, because it tends to be simpler and easier to see what's going on than a module with its own expectations and semantics.

**Reply by qlaffont** [reply]
I will try all your solutions to check what i can do.

In fact i really need to share globals vars across each tests. 

And to setup at the beginning the globals and at the end to close all dbs connections

**Reply by qlaffont** [reply]
@isaacs unfortunately your solutions are not working with my case :/

**Reply by isaacs** [reply]
In that case, I usually just have a utility module at `./test/fixtures/setup.ts` or something, and import it at the start of the test. Sometimes the less clever solution is best.

**Reply by qlaffont** [reply, marked as answer]
This is what i'm currently doing ^^
